### ISSUE TITLE
Sync up to old repo

### ISSUE THREAD

**Issue by Marenthyu** [Marenthyu/smarttv-twitch#1]
Hi there!

The original repo this was forked off off was made private due to the original author not liking my effort to make the app work with the new Twitch API.

I was barely able to collect the relevant issue in some screenshots:

https://lowee.de/chrome_2022-02-10_14-07-44.png
https://lowee.de/chrome_2022-02-10_14-07-55.png
https://lowee.de/chrome_2022-02-10_14-08-02.png
https://lowee.de/chrome_2022-02-10_14-08-10.png
https://lowee.de/chrome_2022-02-10_14-08-21.png
https://lowee.de/chrome_2022-02-10_14-08-29.png
https://lowee.de/chrome_2022-02-10_14-08-37.png
https://lowee.de/chrome_2022-02-10_14-08-46.png
https://lowee.de/chrome_2022-02-10_14-08-53.png
https://lowee.de/chrome_2022-02-10_14-09-01.png
https://lowee.de/chrome_2022-02-10_14-09-09.png
https://lowee.de/chrome_2022-02-10_14-09-15.png
https://lowee.de/chrome_2022-02-10_14-09-20.png
https://lowee.de/chrome_2022-02-10_14-09-27.png
https://lowee.de/chrome_2022-02-10_14-09-33.png

To get the attention back of those involved previously, feel free to use this repository, I'll try my best to make things work in the coming days, even though i currently don't think the one samsung TV i have is compatible - I'll have to check.
@Sellular @r4inX @RaTeNo @Drezus @kaldonir @Yozze85 @georgebvro @toniocarbonia11 @c0ldplasma @lilchina

**Comment by Sellular** [comment]
High key not looking like fgl even wants much help if he keeps acting like he is. Not sure I'd ever want to work with him

**Comment by Marenthyu** [comment]
> High key not looking like fgl even wants much help if he keeps acting like he is. Not sure I'd ever want to work with him

I can understand that sentiment! I still think they want to have things working and once i am not omikron-sick anymore, i sure should be able to figure out the Implicit Authentication Part as well; It does look to me like the majority of it is effectively just a webpage the TV Displays, leaving open _a lot_ of options how to handle things, which we can figure out.

For now, let's not be the Demons we despise / go an eye for an eye and rather try to be the positivity we want to see in them ^_^

**Comment by Drezus** [comment]
> High key not looking like fgl even wants much help if he keeps acting like he is. Not sure I'd ever want to work with him

He's not. I already gave up to the fact I'll have to use Twitch for Xbox. You guys should be considering this alternative too; guy is pretty immature and doesn't like things being made not his own way.

**Comment by Sellular** [comment]
Agreed, it's probably not too bad and we should be able to reuse the
functionality that stores the current refresh token

**Comment by Marenthyu** [comment]
> > High key not looking like fgl even wants much help if he keeps acting like he is. Not sure I'd ever want to work with him
> 
> He's not. I already gave up to the fact I'll have to use Twitch for Xbox. You guys should be considering this alternative too; guy is pretty immature and doesn't like things being made not his own way.

Oh, I have a chromecast myself, which works perfectly fine!
I'm morseo on about the moral stuff of the whole issue as well as wanting the best experience for the end user - Open Source Software is meant to be open and welcoming, which they are actively working against. I'll still try my best to have things be helpful.

**Comment by kaldonir** [comment]
>For now, let's not be the Demons we despise / go an eye for an eye and rather try to be the positivity we want to see in them ^_^

I agree. We're all interested in a working application, and currently it's being maintained by a single individual (and I'm really thankful for that). We should try to be humble here, and accept that the current mainainer doesn't have the time / resources / interest necessary for a collaborative project. 

Maybe it's actually time for a community version of this app. I won't have time to really look into this over the next few weeks, but it seems like we have a small set of people here who aren't completely opposed to that idea.

There was some talk about a discord server in the old (now deleted thread), maybe we should pick that up to start somewhere.

**Comment by Marenthyu** [comment]
> There was some talk about a discord server in the old (now deleted thread), maybe we should pick that up to start somewhere.

I'm not opposed to that!

For now, I see no harm in continuing the discussion here until some actual need for a Discord Server comes up. Otherwise, I am always open for Discussion or to make new channels on my existing one: https://marenthyu.de

**Comment by georgebvro** [comment]
I am very grateful of fgl27's work on the app, I've been using it for years and I think he did a really good job (bar the security issue on which I don't comment since I don't know enough about).

If you guys still want to maintain and port the app to the new API, I reiterate my willingness to help with testing. 

My device:
2015 Samsung UE40JU7000
Tizen 2.3

I think it's the first "modern" version of Tizen so if the app works on this version, it should work on any newer version.

**Comment by Marenthyu** [comment]
That's wonderful to hear @georgebvro ! We'll definitely come back to that offer!

**Comment by r4inX** [comment]
Goddamn, I just got back from work and the repo is private again... Well, just like u guys said, doesn't look like fgl wants help and isnt a very pleasent dude who can discuss this. I will still look at the code, maybe write a own app just for the experience and learning procedure. U guys are cool, I will keep u updated if I get any news. Def gonna check out the discord for quick communication 👍🏻
Have a great evening and stay safe out there

**Comment by Drezus** [comment]
> Honestly I'm not sure how much we'll want to change in regards to the API stuff since it "sounds" like fgl will be working on it to get it working and I don't want to step on any more of his toes.
> 
> I'd say the only thing(s) we should work on here is the OAuth issue and any sort of cleanup of his current code. Sounds like @Marenthyu has a decent grasp of what it would take to move over the OAuth, and personally I don't really want to try to clean up fgl's code 😆 I'll be open for testing but after what happened on the other thread I'm very cautious about trying to touch anything

I'm in the opposite end, then.
I'm just as brazilian as the guy and I don't get why he'd get so overly emotional over that stuff. Specially when it was opensourced.
He's putting pride first and foremost, and if forking his project to work on something *community based* means stepping on his toes for acting like a crybaby, I'm all in for it.

I'm very glad to learn that @Marenthyu doesn't care about the app per-se but rather the end user and everyone depending on this. That's very humble of you and exactly the kind of disposition I'd otherwise expect from him.

Another thing: If the app gets banned, it's not our problem, it's Twitch being the corporate jerks they've always been. I still see absolutely no logic in fgl's fear other than the old "gambiarra" brazilian way of doing things wrong and easy as long as they work.
You can see how much that helped lul

**Comment by Marenthyu** [comment]
I agree, aside from calling Twitch "jerks" - i think saying "They are just defending their existing Terms and Considitons" to be more fitting, but my view may be skewed due to how closely I've been working with them.

Happy to have you on board and thanks for the explanation! I don't quite consider myself all too well-versed in Brazilian culture :P

**Comment by Drezus** [comment]
No prob.
If any of you need help testing, I can help. I have a 49Q60 and didn't find building in Tizen Studio that hard. I'll check the Tizen specs asap.
